src/service/git/git.ts

```diff
@@ -0,0 +1,296 @@
+import simpleGit, { SimpleGit } from "simple-git";
+import { Service } from "typedi";
+import fs from "fs";
+import { LoggerService } from "@bc/service/logger/logger-service";
+import { LoggerServiceFactory } from "@bc/service/logger/logger-service-factory";
+import { GitExecutorResult } from "simple-git/dist/src/lib/types";
+import { NotFoundError } from "@bc/domain/errors";
+import { Octokit } from "@octokit/rest";
+import { OctokitFactory } from "@bc/service/git/octokit";
+
+type GitErrorHandlerFunction = (error: Buffer | Error | undefined, result: Omit<GitExecutorResult, "rejection">) => Buffer | Error | undefined;
+
+@Service()
+export class Git {
+    private readonly logger: LoggerService;
+    private readonly octokit: Octokit;
+
+    constructor () {
+        this.logger = LoggerServiceFactory.getInstance();
+        this.octokit = OctokitFactory.getOctokitInstance();
+    }
+
+    /**
+     * Returns a git instance configured to execute commands in the directory specified or in the current directory
+     * @param cwd [OPTIONAL] working directory on which the git commands will be executed in
+     * @return {SimpleGit} git instance
+     */
+    private git(cwd?: string, errorHandler?: GitErrorHandlerFunction): SimpleGit {
+        if (cwd && errorHandler) {
+            return simpleGit({
+                baseDir: cwd,
+                errors(error, result) {
+                    return errorHandler(error, result);
+                }
+            })
+            .addConfig("user.name", "Github")
+            .addConfig("user.email", "noreply@github.com");
+        } else if (cwd) {
+            return simpleGit({
+                baseDir: cwd,
+            })
+            .addConfig("user.name", "Github")
+            .addConfig("user.email", "noreply@github.com");
+        } else if (errorHandler) {
+            return simpleGit({
+                errors(error, result) {
+                    return errorHandler(error, result);
+                }
+            })
+            .addConfig("user.name", "Github")
+            .addConfig("user.email", "noreply@github.com");
+        }
+        return simpleGit()
+                        .addConfig("user.name", "Github")
+                        .addConfig("user.email", "noreply@github.com");
+    }
+    
+    /**
+     * Returns the git version
+     * @returns {Promise<string>}
+     */
+    async version(): Promise<string | undefined> {
+        const rawOutput = await this.git().raw("version");
+        const match = rawOutput.match(/(\d+\.\d+(\.\d+)?)/);
+        return match && match.length > 1 ? match[1] : undefined;
```

Comment:
it would return back undefined in case match length is less than 1
```suggestion
        return match[1];
```